### PR TITLE
<Stepper/> - most basic stepper functionality

### PR DESCRIPTION
**Pull request by mosheKerbel**
@nitayneeman @shlomitc 
I initialy tried to break into smaller PRs but then they were too dependent
Anyways this PR is the initial one for the Stepper component.
Divided small "features" into commits. Let me know if it is comfortable enough reviewing it this way.

### CODE REVIEW

File: src/Stepper/test/Stepper.spec.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        createDriver(<Stepper activeStep={1} steps={steps} />);
+        expect(consoleErrorSpy).toHaveBeenCalledWith(
+          expect.stringContaining(
+            `Warning: Failed prop: The prop 'steps' at [1] is invalid. Active step can not be disabled. Falling back to 'normal' type for step[1]`,
+          ),
+        );
+      });
+    });
+    describe('Clickable', () => {
+      it('Disabled step should not be clickable', async () => {
```
Comment (nitayneeman) [inline src/Stepper/test/Stepper.spec.js]: Why not wrapping by an internal `describe` instead of attaching it to the test description? 
For example:
```
describe('Disabled step', () => {
  it('should not be clickable', async () => {
})
```

File: src/Stepper/docs/index.story.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+          code({
+            title: 'Stepper',
+            compact: true,
+            source: `<div style={{padding: "20px 20px", backgroundColor: "white"}}>
+              <Stepper activeStep={2} steps={steps} onClick={onClick}/>
+              </div>`,
+          }),
+
+          code({
+            title: 'Stepper',
```
Comment (nitayneeman) [inline src/Stepper/docs/index.story.js]: Probably not part of this PR but should be explained better

File: src/Stepper/dataHooks.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default {
+  step: 'STEP',
```
Comment (nitayneeman) [inline src/Stepper/dataHooks.js]: I think it'd be better to add a prefix of `stepper` to the dataHook, for example: "stepper-step" (relevant for next dataHooks as well)

That's the convention I'm familiar with

File: src/Stepper/Stepper.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  };
+
+  _validateStepType = (steps, activeStep) => {
+    if (
+      activeStep &&
+      typeof activeStep === 'number' &&
+      steps[activeStep].type === STEP_TYPES.DISABLED
+    ) {
+      console.error(
+        `Warning: Failed prop: The prop 'steps' at [${activeStep}] is invalid. Active step can not be disabled. Falling back to 'normal' type for step[${activeStep}]`,
```
Comment (nitayneeman) [inline src/Stepper/Stepper.js]: Not sure if it's a warning or error

File: src/Stepper/Step/StepCircle/StepCircle.st.css [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+.text:disabled > a {
+  color: value(D80);
+}
+
+.text:selected > a {
+  color: value(D80);
+}
+
+.text:disabled > a:hover {
```
Comment (nitayneeman) [inline src/Stepper/Step/StepCircle/StepCircle.st.css]: Redundant?

File: src/Stepper/Stepper.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+  _validateNumberOfSteps(steps) {
+    if (
+      !steps ||
+      !steps.length ||
+      steps.length < MIN_STEPS ||
+      steps.length > MAX_STEPS
+    ) {
+      console.error(
+        `Warning: Failed prop: The prop 'steps' in 'Stepper' has to be an array in the size of ${MIN_STEPS} to ${MAX_STEPS}.`,
```
Comment (nitayneeman) [inline src/Stepper/Stepper.js]: Same here - warning / error?

File: src/Stepper/Step/StepCircle/StepCircle.st.css (hunk 5 above)
Comment (mosheKerbel) [reply src/Stepper/Step/StepCircle/StepCircle.st.css]: actually i must use this state, because i am overriding Text component which has a defenition for `.text > a:hover`

File: src/Stepper/Stepper.js (hunk 6 above)
Comment (mosheKerbel) [reply src/Stepper/Stepper.js]: you are right it should be an error

File: src/Stepper/Stepper.js (hunk 4 above)
Comment (mosheKerbel) [reply src/Stepper/Stepper.js]: you are right it should be an error
